tracing/span.go

```diff
@@ -66,6 +75,22 @@ func newSpan(tracer *Tracer, spanType SpanType) *Span {
 			ZipkinBinaryAnnotationKeyComponent: baseplateComponent,
 		},
 	}
+	if spanType == SpanTypeServer {
+		onServerSpanCreate(span)
+		span.startSpan()
+	}
+	return span
+}
+
+func (s *Span) startSpan() {
+	for _, hook := range s.hooks {
+		hook.OnStart(s)
```

Comment:
Should we check the error here? If not, do we need the interface to actually return an error?

Reply:
I was thinking I would use `s.tracer.Logger` to log an error message?